ui/core/src/api/EthbridgeService/tokenContract.ts

```diff
@@ -0,0 +1,38 @@
+import Web3 from "web3";
+import { AbiItem } from "web3-utils";
+
+// We should add other ABIs as they are required here
```

Comment:
Are these in the parent dir? If so, a  `fetch()` would be most appropriate

Reply:
Before I was mistakenly under the impression that we needed to approve on the bridgebank token which in my head might have been a custom token with custom methods. Since then I found that we actually just need to approve on the erc-20 token associated with the burn or lock and that `approve()` was a general method found as part of the erc-20 spec. This is why I thought it made sense to make this more general and about erc-20 tokens.

I can see some download benefits to using fetch but I think the data we are talking about is pretty small so it probably doesn't matter. We could host the FakeTokenAbis as an asset say `./assets/AliceToken.json` then use fetch to pull it down at runtime. But that doesn't make sense necessarily because AliceToken is not a real token and not part of our mainnet deployment so there is a bit of a hackery type mismatch there. All the tools I see for getting ABIs from the internet are using block explorers and third party tools for published tokens so again hackery. Seems like the cleanest way is to store a general set of abi functions mapped to open zeppelin contracts and only use what we need when we need it which should keep things slim.   

I dunno seems easier to just hard code it here in a neat way we can modularize later perhaps?

Reply:
🏖